google/cloud/bigquery/client.py

```diff
@@ -596,20 +617,35 @@ def create_table(self, table, exists_ok=False, retry=DEFAULT_RETRY, timeout=None
         data = table.to_api_repr()
         try:
             span_attributes = {"path": path, "dataset_id": dataset_id}
-            with create_span(
-                name="BigQuery.createTable", attributes=span_attributes, client=self
-            ):
-                api_response = self._call_api(
-                    retry, method="POST", path=path, data=data, timeout=timeout
-                )
+            api_response = self._call_api(
+                retry,
+                name="BigQuery.createTable",
+                span_attributes=span_attributes,
+                client=self,
+                method="POST",
+                path=path,
+                data=data,
+                timeout=timeout,
+            )
             return Table.from_api_repr(api_response)
         except google.api_core.exceptions.Conflict:
             if not exists_ok:
                 raise
             return self.get_table(table.reference, retry=retry)
 
-    def _call_api(self, retry, **kwargs):
-        call = functools.partial(self._connection.api_request, **kwargs)
+    def _call_api(
+        self,
+        retry,
+        name=None,
+        span_attributes=None,
+        client=None,
+        job_ref=None,
+        **kwargs
+    ):
+        with create_span(
+            name=name, attributes=span_attributes, client=client, job_ref=job_ref
```

Comment:
If `name` is `None`, will this span creation still work? It's a non-optional parameter to `create_span`: https://github.com/googleapis/python-bigquery/blob/fa9f9ca491c3f9954287102c567ec483aa6151d4/google/cloud/bigquery/opentelemetry_tracing.py#L43

Reply:
good point I will need to add some logic for that.